### ISSUE TITLE
discuss: Custom formats are module-scoped

### ISSUE THREAD

**Issue by whitlockjc** [zaggino/z-schema#214]
What was the reasoning behind making `ZSchema.registerFormat` instead of `validator.registerFormat`?  As it stands right now, using multiple `ZSchema` instances in the same process with different custom formats is impossible because the registry of custom formats is stored in the `ZSchema` module itself, not in the `ZSchema` instance.  Before I blindly ask for, or submit a PR to, make the custom formats scoped to the `ZSchema` instance, I wanted to better understand.

**Comment by zaggino** [comment]
Yeah, in this case, that's bad. It probably should have lived on the instance, not the class. I don't see a downside in changing that with a PR.

**Comment by whitlockjc** [comment]
I'll get started.
